src/domain/task-system/services/src/task_executor_impl.rs

```diff
@@ -116,6 +117,36 @@ impl TaskExecutor for TaskExecutorImpl {
                         .clone()
                         .unwrap_or(TaskOutcome::Success(TaskResult::Empty))
                 }
+                LogicalPlan::HardCompactDataset(HardCompactDataset {
+                    dataset_id,
+                    max_slice_size,
+                    max_slice_records,
+                }) => {
+                    let compact_svc = self.catalog.get_one::<dyn CompactService>().int_err()?;
+                    let dataset_repo = self.catalog.get_one::<dyn DatasetRepository>().int_err()?;
+                    let dataset_handle = dataset_repo
+                        .resolve_dataset_ref(&dataset_id.as_local_ref())
+                        .await
+                        .int_err()?;
+
+                    let compact_result = compact_svc
+                        .compact_dataset(
+                            &dataset_handle,
+                            CompactOptions {
+                                max_slice_size: *max_slice_size,
+                                max_slice_records: *max_slice_records,
+                            },
+                            None,
+                        )
+                        .await;
+
+                    match compact_result {
+                        Ok(result) => {
+                            TaskOutcome::Success(TaskResult::CompactDatasetResult(result.into()))
+                        }
+                        Err(_) => TaskOutcome::Failed(TaskError::Empty),
```

Comment:
What kind of error is this? Does it contain any useful details for reporting the result?